workflows/src/test/kotlin/com/r3/corda/lib/tokens/workflows/ConfigSelectionTest.kt

```diff
@@ -0,0 +1,183 @@
+import com.r3.corda.lib.tokens.workflows.internal.selection.MAX_RETRIES_DEFAULT
+import com.r3.corda.lib.tokens.workflows.internal.selection.RETRY_CAP_DEFAULT
+import com.r3.corda.lib.tokens.workflows.internal.selection.RETRY_SLEEP_DEFAULT
+import com.r3.corda.lib.tokens.workflows.internal.selection.TokenSelection
+import com.r3.corda.lib.tokens.workflows.internal.selection.VaultWatcherService
+import com.typesafe.config.ConfigFactory
+import net.corda.core.identity.CordaX500Name
+import net.corda.node.internal.cordapp.TypesafeCordappConfig
+import net.corda.testing.common.internal.testNetworkParameters
+import net.corda.testing.core.TestIdentity
+import net.corda.testing.node.MockServices
+import net.corda.testing.node.createMockCordaService
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.junit.Before
+import org.junit.Test
+
+class ConfigSelectionTest {
+    private lateinit var services: MockServices
+
+    @Before
+    fun setup() {
+        services = MockServices.makeTestDatabaseAndPersistentServices(
+                cordappPackages = listOf("com.r3.corda.lib.tokens.workflows"),
+                initialIdentity = TestIdentity(CordaX500Name("Test", "London", "GB")),
+                networkParameters = testNetworkParameters(minimumPlatformVersion = 4),
+                moreIdentities = emptySet(),
+                moreKeys = emptySet()
+        ).second
+    }
+
+    @Test
+    fun `test full database selection`() {
+        val config = ConfigFactory.parseString("stateSelection {\n" +
+                "database {\n" +
+                "maxRetries: 13\n" +
+                "retrySleep: 300\n" +
+                "retryCap: 1345\n" +
+                "}\n" +
+                "}")
+        val cordappConfig = TypesafeCordappConfig(config)
+        val selection = ConfigSelection.getPreferredSelection(services, cordappConfig)
+        assertThat(selection).isInstanceOf(TokenSelection::class.java)
+        assertThat(selection).hasFieldOrPropertyWithValue("maxRetries", 13)
+        assertThat(selection).hasFieldOrPropertyWithValue("retrySleep", 300)
+        assertThat(selection).hasFieldOrPropertyWithValue("retryCap", 1345)
+    }
+
+    @Test
+    fun `test full in memory selection public key`() {
+        createMockCordaService(services, ::VaultWatcherService)
+        val config = ConfigFactory.parseString("stateSelection {\n" +
+                "inMemory {\n" +
+                "cacheSize: 9000\n" +
+                "indexingStrategy: public_key\n" +
+                "}\n" +
+                "}")
+        val cordappConfig = TypesafeCordappConfig(config)
+
+        val inMemoryConfig = InMemorySelectionConfig.parse(cordappConfig)
+        assertThat(inMemoryConfig.cacheSize).isEqualTo(9000)
+        assertThat(inMemoryConfig.indexingStrategy).isEqualTo(VaultWatcherService.IndexingType.PUBLIC_KEY)
```

Comment:
What does public key indexing do? Seems... pointless?!

Reply:
For now it's the only implemented by Stefano ;) I will raise second PR with external id indexing.